### PR TITLE
DPP-1276 - additional integration tests to show that when an event has an effect…

### PR DESCRIPTION
**Pull request by satya-sanikommu**
…ivefrom later than the effective to, that it will not write the event at all into either DasCommitments or DasCommitmentsHistory

(cherry picked from commit 82d24431cf8b64abe202cf15d34fb04b160e0d4b)

### CODE REVIEW

File: src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs [hunk 1]
```diff
@@ -637,6 +637,100 @@ public void ThenItShouldAddEventPriceHistoryWhereEffectiveFromIsBeforeEffectiveT
             Assert.AreEqual(1, history.Length);
         }
 
+        [Test]
+        public void ThenItShouldNotAddEventWhereEffectiveFromIsBeforeEffectiveTo()
```
Comment (tom-gough) [inline src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs]: looks like the test names are the wrong way around?

File: src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs [hunk 2]
```diff
@@ -637,6 +637,100 @@ public void ThenItShouldAddEventPriceHistoryWhereEffectiveFromIsBeforeEffectiveT
...
+                ProviderId = "369",
+                LearnerId = "258",
+                EmployerAccountId = "2",
+                TrainingType = TrainingTypes.Framework,
+                TrainingId = "854-965-621",
+                TrainingStartDate = new DateTime(2020, 9, 1),
+                TrainingEndDate = new DateTime(2021, 10, 1),
+                TrainingTotalCost = 99999,
+                PaymentStatus = Events.Api.Types.PaymentStatus.Withdrawn,
+                PaymentOrder = 99, //changed
```
Comment (tom-gough) [inline src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs]: what is the purpose of this comment? is it needed?

Comment (davehowlett) [reply src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs]: no - the bones of the test were copied from another and I didn't notice the comment. Will remove it.

File: src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs (hunk 1 above)
Comment (davehowlett) [reply src/Reference/Commitments/SFA.DAS.Payments.Reference.Commitments.IntegrationTests/GivenApiAvailable/WhenReadingEvents.cs]: they certainly seem to be! Code blind when I did that the day before my holiday. I've swapped them around and properly named them

**Review by tom-gough — APPROVED** [review]

**Comment by satya-sanikommu** [comment]
This is only adding an additional integration test to verify the CommitmentsReferenceJob update as part of DPP-1151. No functional changes. Ready to merge

**Pull request merged**